portfolio/src/main/java/com/google/sps/servlets/DeleteCommentServlet.java

```diff
@@ -34,18 +34,52 @@
 import com.google.appengine.api.datastore.Key;
 import com.google.appengine.api.datastore.KeyFactory;
 
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+
+
 import com.google.sps.data.Comment;
+import com.google.sps.data.UserInfo;
 
 /** Servlet responsible for deleting comments. */
 @WebServlet("/delete-comment")
 public class DeleteCommentServlet extends HttpServlet {
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    long id = Long.parseLong(request.getParameter("id"));
+    response.setContentType("application/json");
 
-    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+    long id = Long.parseLong(request.getParameter("id"));
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    datastore.delete(taskEntityKey);
+    Key taskEntityKey = KeyFactory.createKey("Comment", id);
+
+    //Only delete a comment if the author is trying to delete it. 
+    try {
+      String commentUserId = (String) datastore.get(taskEntityKey).getProperty("userId");
+
+      UserService userService = UserServiceFactory.getUserService();
+      UserInfo userInfo = new UserInfo();
+
+      String currentUserId = userService.getCurrentUser().getUserId();
+      userInfo.userId = currentUserId;
+
+      if (userService.isUserLoggedIn()) {
+        if(currentUserId.equals(commentUserId)) {
+          datastore.delete(taskEntityKey);
+        }
+        userInfo.loggedIn = true;
+      } else {
+        String urlToRedirectToAfterUserLogsIn = "/index.html";
+        String loginUrl = userService.createLoginURL(urlToRedirectToAfterUserLogsIn);
+
+        userInfo.loggedIn = false;
+        userInfo.loginUrl = loginUrl;
+      }
+
+      String json = new Gson().toJson(userInfo);
```

Comment:
Why are you returning something from this API call?

It should succeed (500) if the user is able to delete the comment, and fail (400) with an appropriate error message if the user is not logged in or trying to delete a comment they can't delete.

Right now it seems like this API endpoint is trying to do two things -- delete a comment AND check whether a user is logged in.

Reply:
Original motivation was to pass this info back to the delete comment so it could redirect user automatically. 

Rewritten to return just a status instead as suggested.